engine/runtime-instrument-common/src/main/java/org/enso/interpreter/instrument/TypeInfo.java

```diff
@@ -0,0 +1,28 @@
+package org.enso.interpreter.instrument;
+
+import java.util.Arrays;
+
+/**
+ * The type information observed by the instrumentation.
+ *
+ * <p>The list in the type definition represents an intersection type. A list with a single element
+ * represents a simple type.
+ *
+ * @param visibleType the public type of the value visible to the user
+ * @param hiddenType the list of types the value can be converted to
+ */
+public record TypeInfo(String[] visibleType, String[] hiddenType) {
```

Comment:
`record` is immutable structure, but it references mutable `String[]` arrays. Isn't it better to use `List<String>` and make it immutable?